### PR TITLE
Add KBI0012: annex encryption & GPG pinentry

### PR DESCRIPTION
**Pull request by mslw**
This short KBI is converted from my own notes. The origin was an issue on MacOS which was discussed during office hours at some point in March 2023 (and also demonstrated on our cluster).

### CODE REVIEW

File: kbi/0012/index.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.. index::
+   single: git-annex; encryption
+
+KBI0012: Annex encryption & gpg pinentry
```
Comment (adswa) [inline kbi/0012/index.rst:4]: Maybe we can make it more clear in the title that this KBI is about resolving an issue, and not, e.g., about how to set encryption up? Maybe its also useful to cross-reference your usecase in the handbook as a pointer: http://handbook.datalad.org/en/latest/usecases/encrypted_annex.html

Comment (mih) [reply kbi/0012/index.rst:4]: I agree

Comment (mslw) [reply kbi/0012/index.rst]: Good point - updated.

**Comment by jsheunis** [comment]
LGTM!

**Review by christian-monch — APPROVED** [review]
Looks good to me.

**Review by mih — APPROVED** [review]
Works for me, thx!

**Pull request merged**